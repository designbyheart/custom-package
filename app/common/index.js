// @flow
export * from './api-constants'
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
