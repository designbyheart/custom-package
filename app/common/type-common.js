// @flow

export type CustomError = {
  code: string,
  message: string,
}

export const INITIAL_TEST_ACTION: 'INITIAL_TEST_ACTION' = 'INITIAL_TEST_ACTION'

export type InitialTestAction = {
  type: typeof INITIAL_TEST_ACTION,
}

export const initialTestAction = () => ({
  type: INITIAL_TEST_ACTION,
})

export type NavigationParams = {
  [string]: any,
}

export type ReactNavigation = {
  navigation: {
    navigate: (route: string, params?: any) => void,
    state?: {
      params: NavigationParams,
    },
  },
}

export type GenericObject = {
  [string]: any,
}

export type GenericStringObject = {
  [string]: string,
}

export type MessageAnnotation = {
  name: string,
  version: string,
}

export type TopicAnnotation = {
  tid: number,
  mid: number,
}
