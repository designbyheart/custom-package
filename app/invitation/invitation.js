import React, { Component } from "react";
import { View, StyleSheet } from "react-native";
import { Avatar, Text, Icon, Button } from "react-native-elements";
import InvitationText from "./invite-text";
import InvitationActions from "./invitation-actions";

export default class InvitationScreen extends Component {
  static navigationOptions = {
    title: "Accept invitation"
  };

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.inviteContainer}>
          <InvitationText />
        </View>
        <View>
          <InvitationActions />
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    justifyContent: "space-between"
  },
  inviteContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  }
});
