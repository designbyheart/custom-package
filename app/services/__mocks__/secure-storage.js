let secureStorage = {
  ConnectMeKeyChain: {},
}

export const setItem = (key, data) => {
  secureStorage['ConnectMeKeyChain'][key] = data
}

export const getItem = key =>
  Promise.resolve(secureStorage['ConnectMeKeyChain'][key])

export const deleteItem = key => delete secureStorage['ConnectMeKeyChain'][key]
