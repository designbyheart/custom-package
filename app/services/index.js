export * from './api'
export * from './keys'
export * from './secure-storage'
export * from './mapper'
export { default as schemaValidator } from './schema-validator'
export * from './error/error-handler'
