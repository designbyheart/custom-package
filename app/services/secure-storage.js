// @flow
import RNSensitiveInfo from 'react-native-sensitive-info'
import {
  setWalletItem,
  getWalletItem,
  deleteWalletItem,
} from '../bridge/react-native-cxs/RNCxs'

const storageName = {
  sharedPreferencesName: 'ConnectMeSharedPref',
  keychainService: 'ConnectMeKeyChain',
}

export const setItem = (key: string, data: string) =>
  RNSensitiveInfo.setItem(key, data, storageName)

export const getItem = (key: string) =>
  RNSensitiveInfo.getItem(key, storageName)

export const deleteItem = (key: string) =>
  RNSensitiveInfo.deleteItem(key, storageName)

export const setWalletRecordItem = (key: string, data: string) =>
  setWalletItem(key, data)

export const getWalletRecordItem = (key: string) => getWalletItem(key)

export const deleteWalletRecordItem = (key: string) => deleteWalletItem(key)
