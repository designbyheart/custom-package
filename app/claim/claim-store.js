// @flow
import { put, takeLatest, call, all, select } from 'redux-saga/effects'
import type {
  Claim,
  ClaimStore,
  ClaimAction,
  ClaimReceivedAction,
  ClaimStorageFailAction,
  ClaimStorageSuccessAction,
  MapClaimToSenderAction,
  ClaimMap,
  HydrateClaimMapAction,
  HydrateClaimMapFailAction,
  ClaimWithUuid,
} from './type-claim'
import {
  CLAIM_RECEIVED,
  CLAIM_STORAGE_FAIL,
  CLAIM_STORAGE_SUCCESS,
  MAP_CLAIM_TO_SENDER,
  HYDRATE_CLAIM_MAP,
  HYDRATE_CLAIM_MAP_FAIL,
  ERROR_CLAIM_HYDRATE_FAIL,
} from './type-claim'
import type { CustomError } from '../common/type-common'
import { addClaim, getClaim } from '../bridge/react-native-cxs/RNCxs'
import { CLAIM_STORAGE_ERROR } from '../services/error/error-code'
import {
  getConnectionLogoUrl,
  getPoolConfig,
  getClaimMap,
} from '../store/store-selector'
import { setItem, getItem } from '../services/secure-storage'
import { CLAIM_MAP } from '../common/secure-storage-constants'
import { RESET } from '../common/type-common'

export const claimReceived = (claim: Claim): ClaimReceivedAction => ({
  type: CLAIM_RECEIVED,
  claim,
})

export const claimStorageSuccess = (
  messageId: string
): ClaimStorageSuccessAction => ({
  type: CLAIM_STORAGE_SUCCESS,
  messageId,
})

export const claimStorageFail = (
  messageId: string,
  error: CustomError
): ClaimStorageFailAction => ({
  type: CLAIM_STORAGE_FAIL,
  messageId,
  error,
})

export function* claimReceivedSaga(
  action: ClaimReceivedAction
): Generator<*, *, *> {
  try {
    const { claim: { from_did: senderDID, forDID: myPairwiseDid } } = action
    const poolConfig: string = yield select(getPoolConfig)
    const claimFilterJSON: string = yield call(
      addClaim,
      JSON.stringify(action.claim),
      poolConfig
    )

    const claims: Array<ClaimWithUuid> = yield call(
      getClaim,
      claimFilterJSON,
      poolConfig
    )
    const claimMap: ClaimMap = yield select(getClaimMap)

    // There will only be one claim which will be missing from claimMap
    // thats why raising action from loop and it will run only once.
    // We had to do this because api is returning all claims issued by a particular issuer
    for (let c of claims) {
      if (!claimMap[c.claim_uuid]) {
        const logoUrl = yield select(getConnectionLogoUrl, senderDID)
        yield put(
          mapClaimToSender(c.claim_uuid, senderDID, myPairwiseDid, logoUrl)
        )
        yield put(claimStorageSuccess(action.claim.messageId))

        // persist claimMap to secure storage
        // TODO:PS: replace with fork redux-effect
        Object.assign(claimMap, {
          [c.claim_uuid]: {
            senderDID,
            myPairwiseDid,
            logoUrl,
          },
        })

        yield call(setItem, CLAIM_MAP, JSON.stringify(claimMap))
        break
      }
    }
  } catch (e) {
    // we got error while saving claim in wallet, what to do now?
    yield put(claimStorageFail(action.claim.messageId, CLAIM_STORAGE_ERROR(e)))
  }
}

export const mapClaimToSender = (
  claimUuid: string,
  senderDID: string,
  myPairwiseDID: string,
  logoUrl: string
): MapClaimToSenderAction => ({
  type: MAP_CLAIM_TO_SENDER,
  claimUuid,
  senderDID,
  myPairwiseDID,
  logoUrl,
})

export const hydrateClaimMap = (claimMap: ClaimMap) => ({
  type: HYDRATE_CLAIM_MAP,
  claimMap,
})

export const hydrateClaimMapFail = (error: CustomError) => ({
  type: HYDRATE_CLAIM_MAP_FAIL,
  error,
})

export function* hydrateClaimMapSaga(): Generator<*, *, *> {
  try {
    const fetchedClaimMap = yield call(getItem, CLAIM_MAP)
    if (fetchedClaimMap) {
      const claimMap: ClaimMap = JSON.parse(fetchedClaimMap)
      yield put(hydrateClaimMap(claimMap))
    }
  } catch (e) {
    yield put(
      hydrateClaimMapFail({
        code: ERROR_CLAIM_HYDRATE_FAIL.code,
        message: `${ERROR_CLAIM_HYDRATE_FAIL.message}:${e.message}`,
      })
    )
  }
}

export function* watchClaimReceived(): any {
  yield takeLatest(CLAIM_RECEIVED, claimReceivedSaga)
}

export function* watchClaim(): Generator<*, *, *> {
  yield all([watchClaimReceived()])
}

const initialState = { claimMap: {} }

export default function claimReducer(
  state: ClaimStore = initialState,
  action: ClaimAction
) {
  switch (action.type) {
    case CLAIM_RECEIVED:
      return {
        ...state,
        [action.claim.messageId]: {
          claim: action.claim,
        },
      }

    case CLAIM_STORAGE_FAIL:
      return {
        ...state,
        [action.messageId]: {
          ...state[action.messageId],
          error: action.error,
        },
      }

    case CLAIM_STORAGE_SUCCESS: {
      const { [action.messageId]: deleted, ...newState } = state
      return newState
    }

    case MAP_CLAIM_TO_SENDER:
      const { claimUuid, senderDID, myPairwiseDID, logoUrl } = action
      return {
        ...state,
        claimMap: {
          ...state.claimMap,
          [claimUuid]: {
            senderDID,
            myPairwiseDID,
            logoUrl,
          },
        },
      }

    case HYDRATE_CLAIM_MAP:
      return {
        ...state,
        claimMap: action.claimMap,
      }

    case RESET:
      return initialState

    default:
      return state
  }
}
