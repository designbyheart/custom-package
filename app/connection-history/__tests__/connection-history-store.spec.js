// @flow
import {
  all,
  takeLatest,
  takeEvery,
  put,
  call,
  select,
} from 'redux-saga/effects'
import connectionHistoryReducer, {
  loadHistory,
  loadHistorySuccess,
  loadHistoryFail,
  loadHistorySaga,
  recordHistoryEvent,
  historyEventOccurred,
  historyEventOccurredSaga,
  convertInvitationToHistoryEvent,
  convertConnectionSuccessToHistoryEvent,
  convertClaimOfferToHistoryEvent,
  convertProofRequestToHistoryEvent,
  convertProofAutoFillToHistoryEvent,
} from '../connection-history-store'
import { initialTestAction } from '../../common/type-common'
import {
  getTestInvitationPayload,
  successConnectionData,
  claimOffer,
  claimRequest,
  claim,
  proofRequest,
  senderDid1,
  fulfilledRequestedAttributes,
} from '../../../__mocks__/static-data'
import { invitationRejected } from '../../invitation/invitation-store'
import { saveNewConnection } from '../../store/connections-store'
import {
  claimOfferReceived,
  claimRequestSuccess,
} from '../../claim-offer/claim-offer-store'
import { claimReceived } from '../../claim/claim-store'
import {
  proofRequestReceived,
  sendProof,
  proofRequestAutoFill,
} from '../../proof-request/proof-request-store'
import {
  HISTORY_EVENT_STORAGE_KEY,
  ERROR_LOADING_HISTORY,
  ERROR_HISTORY_EVENT_OCCURED,
} from '../type-connection-history'
import { getProofRequest } from '../../store/store-selector'
import { getItem } from '../../services/secure-storage'
import {
  invitationReceivedEvent,
  newConnectionSuccessEvent,
  claimOfferReceivedEvent,
  proofRequestReceivedEvent,
  proofRequestAutofillEvent,
  proofRequestAutofill,
} from '../../../__mocks__/static-data'

jest.mock('../../services/uuid')

function getHistoryData() {
  // generate history data from static data that we have

  const invitationGenerator = getTestInvitationPayload()
  const invitationPayload = invitationGenerator.next().value
  let sender1History = []
  if (invitationPayload) {
    // add history for connection request
    sender1History.push(
      convertInvitationToHistoryEvent(invitationPayload.payload)
    )
  }

  // add history for connection success
  sender1History.push(
    convertConnectionSuccessToHistoryEvent(
      saveNewConnection(successConnectionData)
    )
  )

  // add history for claim offer
  sender1History.push(
    convertClaimOfferToHistoryEvent(
      claimOfferReceived(claimOffer.payload, claimOffer.payloadInfo)
    )
  )

  // add history for proof request
  sender1History.push(
    convertProofRequestToHistoryEvent(
      proofRequestReceived(proofRequest.payload, proofRequest.payloadInfo)
    )
  )

  // add history for proof sent
  sender1History.push(
    convertProofAutoFillToHistoryEvent(
      proofRequestAutoFill(
        proofRequest.payloadInfo.uid,
        fulfilledRequestedAttributes
      ),
      proofRequest.payload.data.name,
      proofRequest.payloadInfo.remotePairwiseDID
    )
  )

  return {
    [senderDid1]: sender1History,
  }
}

describe('Store: ConnectionHistory', () => {
  let initialState
  beforeEach(() => {
    initialState = connectionHistoryReducer(undefined, initialTestAction())
  })

  it('match loadHistory', () => {
    expect(
      connectionHistoryReducer(initialState, loadHistory())
    ).toMatchSnapshot()
  })

  it('match loadHistorySuccess', () => {
    expect(
      connectionHistoryReducer(
        initialState,
        loadHistorySuccess(getHistoryData())
      )
    ).toMatchSnapshot()
  })

  it('match loadHistoryFail', () => {
    expect(
      connectionHistoryReducer(
        initialState,
        loadHistoryFail({
          code: 'TEST 101',
          message: 'Load History failed',
        })
      )
    ).toMatchSnapshot()
  })

  it('match record history event', () => {
    expect(
      connectionHistoryReducer(
        initialState,
        recordHistoryEvent(
          convertClaimOfferToHistoryEvent(
            claimOfferReceived(claimOffer.payload, claimOffer.payloadInfo)
          )
        )
      )
    ).toMatchSnapshot()
  })

  it('historyEventOccurredSaga should raise success for correct invitation received ', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(invitationReceivedEvent)
    )
    historyEvent = convertInvitationToHistoryEvent(
      invitationReceivedEvent.data.payload
    )

    expect(gen.next().value).toEqual(put(recordHistoryEvent(historyEvent)))
  })

  it('historyEventOccurredSaga should raise success for correct new connection ', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(newConnectionSuccessEvent)
    )
    historyEvent = convertConnectionSuccessToHistoryEvent(
      newConnectionSuccessEvent
    )

    expect(gen.next().value).toEqual(put(recordHistoryEvent(historyEvent)))
  })

  it('historyEventOccurredSaga should raise success for correct claim offer received', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(claimOfferReceivedEvent)
    )
    historyEvent = convertClaimOfferToHistoryEvent(claimOfferReceivedEvent)
    expect(gen.next().value).toEqual(put(recordHistoryEvent(historyEvent)))
  })

  it('historyEventOccurredSaga should raise success for correct proof request received', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(proofRequestReceivedEvent)
    )
    historyEvent = convertProofRequestToHistoryEvent(proofRequestReceivedEvent)
    expect(gen.next().value).toEqual(put(recordHistoryEvent(historyEvent)))
  })

  it('historyEventOccurredSaga should raise success for proof autofill', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(proofRequestAutofillEvent)
    )
    expect(gen.next().value).toEqual(
      select(getProofRequest, proofRequestAutofillEvent.uid)
    )
    historyEvent = convertProofAutoFillToHistoryEvent(
      proofRequestAutofillEvent,
      proofRequestAutofill.originalProofRequestData.name,
      proofRequestAutofill.remotePairwiseDID
    )
  })

  it('historyEventOccuredSaga should raise failure in case anything fails', () => {
    const gen = historyEventOccurredSaga(
      historyEventOccurred(proofRequestAutofillEvent)
    )
    expect(gen.next().value).toEqual(
      select(getProofRequest, proofRequestAutofillEvent.uid)
    )
    const error = new Error()
    expect(gen.throw(error).value).toEqual(
      put(
        loadHistoryFail({
          ...ERROR_HISTORY_EVENT_OCCURED,
          message: `${ERROR_HISTORY_EVENT_OCCURED.message} ${error.message}`,
        })
      )
    )
  })

  it('convertInvitationToHistoryEvent should raise success', () => {
    expect(
      convertInvitationToHistoryEvent(invitationReceivedEvent.data.payload)
    ).toMatchSnapshot()
  })

  it('convertConnectionSuccessToHistoryEvent should raise success', () => {
    expect(
      convertConnectionSuccessToHistoryEvent(newConnectionSuccessEvent)
    ).toMatchSnapshot()
  })
  it('convertClaimOfferToHistoryEvent should raise success', () => {
    expect(
      convertClaimOfferToHistoryEvent(claimOfferReceivedEvent)
    ).toMatchSnapshot()
  })

  it('convertProofRequestToHistoryEvent should raise success', () => {
    expect(
      convertProofRequestToHistoryEvent(proofRequestReceivedEvent)
    ).toMatchSnapshot()
  })
  it('convertProofAutoFillToHistoryEvent should raise success', () => {
    expect(
      convertProofAutoFillToHistoryEvent(
        proofRequestAutofillEvent,
        proofRequestAutofill.originalProofRequestData.name,
        proofRequestAutofill.remotePairwiseDID
      )
    ).toMatchSnapshot()
  })

  it('should raise failure in case data getItem fails', () => {
    const gen = loadHistorySaga()
    expect(gen.next().value).toEqual(call(getItem, HISTORY_EVENT_STORAGE_KEY))
    const error = new Error()
    expect(gen.throw(error).value).toEqual(
      put(
        loadHistoryFail({
          ...ERROR_LOADING_HISTORY,
          message: `${ERROR_LOADING_HISTORY.message} ${error.message}`,
        })
      )
    )
    expect(gen.next().done).toBe(true)
  })
})
