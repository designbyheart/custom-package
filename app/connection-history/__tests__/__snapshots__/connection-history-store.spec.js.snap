// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store: ConnectionHistory convertClaimOfferToHistoryEvent should raise success 1`] = `
Object {
  "action": "RECEIVED",
  "data": Array [
    Object {
      "data": "Address Address Address",
      "label": "Address 1",
    },
    Object {
      "data": "Address 2 Address 2 Address 2",
      "label": "Address 2",
    },
  ],
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "Home Address",
  "originalPayload": Object {
    "payload": Object {
      "data": Object {
        "claimDefinitionSchemaSequenceNumber": 36,
        "name": "Home Address",
        "revealedAttributes": Array [
          Object {
            "data": "Address Address Address",
            "label": "Address 1",
          },
          Object {
            "data": "Address 2 Address 2 Address 2",
            "label": "Address 2",
          },
        ],
        "version": "1.0.0",
      },
      "issuer": Object {
        "did": "issuerDid",
        "name": "Test Issuer",
      },
      "statusMsg": "pending",
    },
    "payloadInfo": Object {
      "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
      "senderLogoUrl": "http://testissuer.com/logoUrl.png",
      "uid": "usd123",
    },
    "type": "CLAIM_OFFER_RECEIVED",
  },
  "remoteDid": "ha66899sadfjZJGINKN0770",
  "status": "RECEIVED",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "CLAIM",
}
`;

exports[`Store: ConnectionHistory convertConnectionSuccessToHistoryEvent should raise success 1`] = `
Object {
  "action": "CONNECTED",
  "data": Array [
    Object {
      "data": "2018-01-10T04:32:43+05:30",
      "label": "Established On",
    },
  ],
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "sender1",
  "originalPayload": Object {
    "connection": Object {
      "identifier": "pairwiseIdentifier1",
      "logoUrl": "lu",
      "myPairwiseAgentDid": "myPairwiseAgentDID",
      "myPairwiseAgentVerKey": "myPairwiseAgentVerKey",
      "myPairwiseDid": "pairwiseIdentifier1",
      "myPairwisePeerVerKey": "senderVerificationKey1",
      "myPairwiseVerKey": "pairwiseVerificationKey1",
      "requestId": "requestId1",
      "senderAgencyDetail": Object {
        "DID": "senderDID1",
        "endpoint": "endpoint",
        "verKey": "senderVerificationKey1",
      },
      "senderAgentKeyDelegationProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "senderDID": "senderDID1",
      "senderDetail": Object {
        "DID": "senderDID1",
        "agentKeyDlgProof": Object {
          "agentDID": "senderAgentDID1",
          "agentDelegatedKey": "agentDelegatedKey",
          "signature": "signature",
        },
        "logoUrl": "http://testissuer.com/logoUrl.png",
        "name": "sender1",
        "verKey": "senderVerificationKey1",
      },
      "senderEndpoint": "endpoint",
      "senderLogoUrl": "lu",
      "senderName": "sender1",
      "senderVerificationKey": "sVk",
      "targetName": "target name",
    },
    "type": "NEW_CONNECTION_SUCCESS",
  },
  "remoteDid": "senderDID1",
  "status": "CONNECTED",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "INVITATION",
}
`;

exports[`Store: ConnectionHistory convertInvitationToHistoryEvent should raise success 1`] = `
Object {
  "action": "CONNECTION REQUEST",
  "data": Object {},
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "sender1",
  "originalPayload": Object {
    "requestId": "requestId1",
    "senderAgencyDetail": Object {
      "DID": "senderDID1",
      "endpoint": "endpoint",
      "verKey": "senderVerificationKey1",
    },
    "senderAgentKeyDelegationProof": Object {
      "agentDID": "senderAgentDID1",
      "agentDelegatedKey": "agentDelegatedKey",
      "signature": "signature",
    },
    "senderDID": "senderDID1",
    "senderDetail": Object {
      "DID": "senderDID1",
      "agentKeyDlgProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "logoUrl": "http://testissuer.com/logoUrl.png",
      "name": "sender1",
      "verKey": "senderVerificationKey1",
    },
    "senderEndpoint": "endpoint",
    "senderLogoUrl": "lu",
    "senderName": "sender1",
    "senderVerificationKey": "sVk",
    "targetName": "target name",
  },
  "remoteDid": "senderDID1",
  "status": "CONNECTION REQUEST",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "INVITATION",
}
`;

exports[`Store: ConnectionHistory convertProofAutoFillToHistoryEvent should raise success 1`] = `
Object {
  "action": "SHARED",
  "data": Array [
    Object {
      "data": "address1",
      "label": "address1",
    },
    Object {
      "data": "address2",
      "label": "address2",
    },
  ],
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "proof_req_1",
  "originalPayload": Object {
    "requestedAttributes": Array [
      Object {
        "data": "address1",
        "label": "address1",
      },
      Object {
        "data": "address2",
        "label": "address2",
      },
    ],
    "type": "PROOF_REQUEST_AUTO_FILL",
    "uid": "pid123",
  },
  "remoteDid": "remotePairWiseDID",
  "status": "SHARED",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "PROOF",
}
`;

exports[`Store: ConnectionHistory convertProofRequestToHistoryEvent should raise success 1`] = `
Object {
  "action": "PROOF RECEIVED",
  "data": Array [
    Object {
      "label": "Address 1",
    },
    Object {
      "label": "Address 2",
    },
  ],
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "Home Address",
  "originalPayload": Object {
    "payload": Object {
      "data": Object {
        "name": "Home Address",
        "requestedAttributes": Array [
          Object {
            "label": "Address 1",
          },
          Object {
            "label": "Address 2",
          },
        ],
        "version": "1.0.0",
      },
      "originalProofRequestData": Object {
        "name": "proof_req_1",
        "nonce": "123432421212",
        "requested_attrs": Object {
          "attr1_uuid": Object {
            "issuer_did": "issuerDid",
            "name": "address1",
            "schema_seq_no": 36,
          },
          "attr2_uuid": Object {
            "issuer_did": "issuerDid",
            "name": "address2",
            "schema_seq_no": 36,
          },
        },
        "requested_predicates": Object {},
        "version": "0.1",
      },
      "requester": Object {
        "name": "Test Issuer",
      },
      "statusMsg": "pending",
    },
    "payloadInfo": Object {
      "remotePairwiseDID": "senderDID1",
      "senderLogoUrl": "http://testissuer.com/logoUrl.png",
      "uid": "pid123",
    },
    "type": "PROOF_REQUEST_RECEIVED",
  },
  "remoteDid": "senderDID1",
  "status": "PROOF RECEIVED",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "PROOF",
}
`;

exports[`Store: ConnectionHistory match loadHistory 1`] = `
Object {
  "data": null,
  "error": null,
  "isLoading": true,
}
`;

exports[`Store: ConnectionHistory match loadHistoryFail 1`] = `
Object {
  "data": null,
  "error": Object {
    "code": "TEST 101",
    "message": "Load History failed",
  },
  "isLoading": false,
}
`;

exports[`Store: ConnectionHistory match loadHistorySuccess 1`] = `
Object {
  "data": Object {
    "senderDID1": Array [
      Object {
        "action": "CONNECTION REQUEST",
        "data": Object {},
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "sender1",
        "originalPayload": Object {
          "requestId": "requestId1",
          "senderAgencyDetail": Object {
            "DID": "senderDID1",
            "endpoint": "endpoint",
            "verKey": "senderVerificationKey1",
          },
          "senderAgentKeyDelegationProof": Object {
            "agentDID": "senderAgentDID1",
            "agentDelegatedKey": "agentDelegatedKey",
            "signature": "signature",
          },
          "senderDID": "senderDID1",
          "senderDetail": Object {
            "DID": "senderDID1",
            "agentKeyDlgProof": Object {
              "agentDID": "senderAgentDID1",
              "agentDelegatedKey": "agentDelegatedKey",
              "signature": "signature",
            },
            "logoUrl": "http://testissuer.com/logoUrl.png",
            "name": "sender1",
            "verKey": "senderVerificationKey1",
          },
          "senderEndpoint": "endpoint",
          "senderLogoUrl": "lu",
          "senderName": "sender1",
          "senderVerificationKey": "sVk",
          "targetName": "target name",
        },
        "remoteDid": "senderDID1",
        "status": "CONNECTION REQUEST",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "INVITATION",
      },
      Object {
        "action": "CONNECTED",
        "data": Array [
          Object {
            "data": "2018-01-10T04:32:43+05:30",
            "label": "Established On",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": undefined,
        "originalPayload": Object {
          "connection": Object {
            "newConnection": Object {
              "identifier": "pairwiseIdentifier1",
              "logoUrl": "lu",
              "myPairwiseAgentDid": "myPairwiseAgentDID",
              "myPairwiseAgentVerKey": "myPairwiseAgentVerKey",
              "myPairwiseDid": "pairwiseIdentifier1",
              "myPairwisePeerVerKey": "senderVerificationKey1",
              "myPairwiseVerKey": "pairwiseVerificationKey1",
              "requestId": "requestId1",
              "senderAgencyDetail": Object {
                "DID": "senderDID1",
                "endpoint": "endpoint",
                "verKey": "senderVerificationKey1",
              },
              "senderAgentKeyDelegationProof": Object {
                "agentDID": "senderAgentDID1",
                "agentDelegatedKey": "agentDelegatedKey",
                "signature": "signature",
              },
              "senderDID": "senderDID1",
              "senderDetail": Object {
                "DID": "senderDID1",
                "agentKeyDlgProof": Object {
                  "agentDID": "senderAgentDID1",
                  "agentDelegatedKey": "agentDelegatedKey",
                  "signature": "signature",
                },
                "logoUrl": "http://testissuer.com/logoUrl.png",
                "name": "sender1",
                "verKey": "senderVerificationKey1",
              },
              "senderEndpoint": "endpoint",
              "senderLogoUrl": "lu",
              "senderName": "sender1",
              "senderVerificationKey": "sVk",
              "targetName": "target name",
            },
          },
          "type": "NEW_CONNECTION",
        },
        "remoteDid": undefined,
        "status": "CONNECTED",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "INVITATION",
      },
      Object {
        "action": "RECEIVED",
        "data": Array [
          Object {
            "data": "Address Address Address",
            "label": "Address 1",
          },
          Object {
            "data": "Address 2 Address 2 Address 2",
            "label": "Address 2",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "Home Address",
        "originalPayload": Object {
          "payload": Object {
            "data": Object {
              "claimDefinitionSchemaSequenceNumber": 36,
              "name": "Home Address",
              "revealedAttributes": Array [
                Object {
                  "data": "Address Address Address",
                  "label": "Address 1",
                },
                Object {
                  "data": "Address 2 Address 2 Address 2",
                  "label": "Address 2",
                },
              ],
              "version": "1.0.0",
            },
            "issuer": Object {
              "did": "issuerDid",
              "name": "Test Issuer",
            },
            "statusMsg": "pending",
          },
          "payloadInfo": Object {
            "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
            "senderLogoUrl": "http://testissuer.com/logoUrl.png",
            "uid": "usd123",
          },
          "type": "CLAIM_OFFER_RECEIVED",
        },
        "remoteDid": "ha66899sadfjZJGINKN0770",
        "status": "RECEIVED",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "CLAIM",
      },
      Object {
        "action": "PROOF RECEIVED",
        "data": Array [
          Object {
            "label": "Address 1",
          },
          Object {
            "label": "Address 2",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "Home Address",
        "originalPayload": Object {
          "payload": Object {
            "data": Object {
              "name": "Home Address",
              "requestedAttributes": Array [
                Object {
                  "label": "Address 1",
                },
                Object {
                  "label": "Address 2",
                },
              ],
              "version": "1.0.0",
            },
            "originalProofRequestData": Object {
              "name": "proof_req_1",
              "nonce": "123432421212",
              "requested_attrs": Object {
                "attr1_uuid": Object {
                  "issuer_did": "issuerDid",
                  "name": "address1",
                  "schema_seq_no": 36,
                },
                "attr2_uuid": Object {
                  "issuer_did": "issuerDid",
                  "name": "address2",
                  "schema_seq_no": 36,
                },
              },
              "requested_predicates": Object {},
              "version": "0.1",
            },
            "requester": Object {
              "name": "Test Issuer",
            },
            "statusMsg": "pending",
          },
          "payloadInfo": Object {
            "remotePairwiseDID": "senderDID1",
            "senderLogoUrl": "http://testissuer.com/logoUrl.png",
            "uid": "pid123",
          },
          "type": "PROOF_REQUEST_RECEIVED",
        },
        "remoteDid": "senderDID1",
        "status": "PROOF RECEIVED",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "PROOF",
      },
      Object {
        "action": "SHARED",
        "data": Array [
          Object {
            "data": "Address 1",
            "label": "address1",
          },
          Object {
            "data": "Address 2",
            "label": "address2",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "Home Address",
        "originalPayload": Object {
          "requestedAttributes": Array [
            Object {
              "data": "Address 1",
              "label": "address1",
            },
            Object {
              "data": "Address 2",
              "label": "address2",
            },
          ],
          "type": "PROOF_REQUEST_AUTO_FILL",
          "uid": "pid123",
        },
        "remoteDid": "senderDID1",
        "status": "SHARED",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "PROOF",
      },
    ],
  },
  "error": null,
  "isLoading": false,
}
`;

exports[`Store: ConnectionHistory match record history event 1`] = `
Object {
  "data": Object {
    "ha66899sadfjZJGINKN0770": Array [
      Object {
        "action": "RECEIVED",
        "data": Array [
          Object {
            "data": "Address Address Address",
            "label": "Address 1",
          },
          Object {
            "data": "Address 2 Address 2 Address 2",
            "label": "Address 2",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "Home Address",
        "originalPayload": Object {
          "payload": Object {
            "data": Object {
              "claimDefinitionSchemaSequenceNumber": 36,
              "name": "Home Address",
              "revealedAttributes": Array [
                Object {
                  "data": "Address Address Address",
                  "label": "Address 1",
                },
                Object {
                  "data": "Address 2 Address 2 Address 2",
                  "label": "Address 2",
                },
              ],
              "version": "1.0.0",
            },
            "issuer": Object {
              "did": "issuerDid",
              "name": "Test Issuer",
            },
            "statusMsg": "pending",
          },
          "payloadInfo": Object {
            "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
            "senderLogoUrl": "http://testissuer.com/logoUrl.png",
            "uid": "usd123",
          },
          "type": "CLAIM_OFFER_RECEIVED",
        },
        "remoteDid": "ha66899sadfjZJGINKN0770",
        "status": "RECEIVED",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "CLAIM",
      },
    ],
  },
  "error": null,
  "isLoading": false,
}
`;
